google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/PostgreSQLStatementParser.java

```diff
@@ -92,6 +108,10 @@ String removeCommentsAndTrimInternal(String sql) {
           continue;
         } else if (sql.length() > index + 1 && c == SLASH && sql.charAt(index + 1) == ASTERISK) {
           multiLineCommentLevel++;
+          if (index >= 1) {
+            whitespaceBeforeOrAfterMultiLineComment = Character.isWhitespace(sql.charAt(index - 1));
+          }
+          multiLineCommentStartIdx = index;
```

Comment:
I think this would not go wrong. We are checking for space before/after only for the outermost comment. I've added a test case having nested comments too in that regard.